src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -73,357 +56,214 @@ public AnnotationConfiguration LoadConfig()
+                // TODO: return InternalServerError for common Exception and Forbidden for PasswordProtectedException
+                return Request.CreateResponse(HttpStatusCode.Forbidden, new Resources().GenerateException(ex, password));
             }
-        }      
+        }
 
-        /// <summary>
-        /// Get document page
-        /// </summary>
-        /// <param name="loadDocumentPageRequest"></param>
-        /// <returns>Document page image</returns>
-        [HttpPost]
-        [Route("loadDocumentPage")]
-        public HttpResponseMessage LoadDocumentPage(AnnotationPostedDataEntity loadDocumentPageRequest)
+        public AnnotatedDocumentEntity LoadDocument(AnnotationPostedDataEntity loadDocumentRequest)
         {
-            string password = "";
-            try
+            string password = loadDocumentRequest.password;
+
+            // check if document contains annotations
+          
+            // initiate pages description list
+            // initiate custom Document description object
+            AnnotatedDocumentEntity description = new AnnotatedDocumentEntity();
+
+            string documentGuid = loadDocumentRequest.guid;
+
+            using (FileStream outputStream = File.OpenRead(documentGuid))
             {
-                // get/set parameters
-                string documentGuid = loadDocumentPageRequest.guid;
-                int pageNumber = loadDocumentPageRequest.page;
-                password = loadDocumentPageRequest.password;
-                PageDescriptionEntity loadedPage = new PageDescriptionEntity();
-                ImageOptions imageOptions = new ImageOptions()
-                {
-                    PageNumber = pageNumber,
-                    CountPagesToConvert = 1,
-                    Password = password
-                };
-                string documentPath = GetDocumentPath(documentGuid);               
-                // get page image
-                byte[] bytes;
-                using (MemoryStream memoryStream = new MemoryStream())
+                using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(outputStream, GetLoadOptions(password)))
                 {
-                    using (Stream document = File.Open(documentGuid, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                    IDocumentInfo info = annotator.Document.GetDocumentInfo();
+
+                    AnnotationBase[] annotations = GetAnnotations(loadDocumentRequest.guid, info.FileType.ToString(), password);
+
+                    description.guid = loadDocumentRequest.guid;
+                    string documentType = SupportedImageFormats.Contains(info.FileType.Extension) ? "image" : info.FileType.ToString();
+                    description.supportedAnnotations = new SupportedAnnotations().GetSupportedAnnotations(documentType);
+
+                    List<string> pagesContent = new List<string>();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'pagesContent'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)